packages/dx-react-core/src/plugged/template-connector.jsx

```diff
@@ -19,31 +19,40 @@ export class TemplateConnector extends React.Component {
   componentWillMount() {
     const { pluginHost } = this.context;
     pluginHost.registerSubscription(this.subscription);
+
+    this.prepareForRendering();
+  }
+  componentWillReceiveProps(props) {
+    this.prepareForRendering(props);
   }
   componentWillUnmount() {
     const { pluginHost } = this.context;
     pluginHost.unregisterSubscription(this.subscription);
   }
+  prepareForRendering(props) {
+    const { children } = props || this.props;
+    const { pluginHost } = this.context;
+
+    const { getters, trackedDependencies } = getAvailableGetters(pluginHost);
+    this.trackedDependencies = trackedDependencies;
+    const actions = getAvailableActions(pluginHost);
+
+    this.setState({ children: children(getters, actions) });
+  }
   updateConnection() {
     const { pluginHost } = this.context;
 
     if (isTrackedDependenciesChanged(pluginHost, this.trackedDependencies)) {
+      this.prepareForRendering();
       this.forceUpdate();
     }
   }
   render() {
-    const { children } = this.props;
-    const { pluginHost } = this.context;
-
-    const { getters, trackedDependencies } = getAvaliableGetters(pluginHost);
-    this.trackedDependencies = trackedDependencies;
-    const actions = getAvaliableActions(pluginHost);
-
-    return children(getters, actions);
+    return this.state.children;
   }
 }
-
 TemplateConnector.propTypes = {
+  // eslint-disable-next-line react/no-unused-prop-types
```

Comment:
I prefer to write 'props.children || this.props.children' in the prepareForRendering method instead.